Update to master

johnnybatesjr opened the pull request:
## Fixes # .

## Summary/Motivation:


## Changes proposed in this PR:
-
-

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the BSD license.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

qtothec commented:
We made a few edits to the MC++ interface, so it looks like you have a few merge conflicts. If you'd like some help getting them sorted, I can work out a time with you.

whart222 committed:
Changing test category

blnicho committed:
Merge branch 'master' into finalize-release

blnicho committed:
More updates to the CHANGELOG in preparation for the 5.7.3 release

jsiirola committed:
Merge pull request #1794 from carldlaird/units-performance

Improved units processing performance

jsiirola committed:
Python 2.x compatibility

Rework logic for closing the underlying transport object to work in
Python 2 and 3 using public API methods.

jsiirola committed:
The SolverManager_NEOS does not need to close the kestrelAMPL transport

jsiirola committed:
Removing debugging code

jsiirola committed:
Reorganize NEOS tests

jsiirola committed:
Further standardization of NEOS tests

jsiirola committed:
Add writer consistency checks between filenames and formats

jsiirola committed:
Specify results format

jsiirola committed:
Add a PR job to test the full NEOS solver interface

jsiirola committed:
More rigorous comparison when checking units

blnicho committed:
Removing deprecated functionality from GAMS tests

blnicho committed:
A few minor changes to fix failing tests

blnicho committed:
Always default to yaml when writing results even when yaml is unavailable

jsiirola committed:
Improve robustness of pint unit comparisons

jsiirola committed:
Updating neos test models

jsiirola committed:
Promote _equivalent_{pint_units,to_dimensionless} to units cointainer

jsiirola committed:
Merge pull request #1804 from jsiirola/fname-format-consistency

Add writer consistency checks between filenames and formats

jsiirola committed:
Merge branch 'master' into neos

jsiirola committed:
Relax NEOS test solution tolerances

blnicho committed:
Merge branch 'master' into finalize-release

blnicho committed:
More updates to the CHANGELOG in preparation for the 5.7.3 release

jsiirola committed:
Update expected result from neos.raposa

blnicho committed:
Merge pull request #1805 from jsiirola/units-comp-fix

Improve robustness of pint units comparisons

mrmundt committed:
Initialize self.tempdir in TempfileManagerClass

blnicho committed:
NFC: Fixing typos in comments

blnicho committed:
Merge pull request #1802 from whart222/neos

Resolving NEOS interface issues

blnicho committed:
Merge pull request #1806 from mrmundt/tempfile-bug

Initialize self.tempdir in TempfileManagerClass

blnicho committed:
Merge branch 'master' into finalize-release

blnicho committed:
More updates to the CHANGELOG in preparation for the 5.7.3 release

jsiirola committed:
Updating changelog

jsiirola committed:
Finalizing the Pyomo 5.7.3 release files

blnicho committed:
Merge pull request #1807 from Pyomo/finalize-release

Finalizing the 5.7.3 release

jsiirola committed:
Preparing master for development (6.0.0.dev0)

blnicho committed:
Merge pull request #1808 from Pyomo/finalize-release

Preparing master for development (6.0.0.dev0)

carldlaird committed:
removing model supports method

carldlaird committed:
cleaning up tests

carldlaird committed:
cleaning up the handling of no hessian support

carldlaird committed:
adding test skips for ipopt not present

carldlaird committed:
merge conflict in cyipopt_solver.py

mrmundt committed:
Merge branch 'master' into remove_py27_gha

carldlaird committed:
fixing six import

blnicho committed:
Merge pull request #1763 from mrmundt/remove_py27_gha

Remove Python 2.7, 3.5, and pypy2 from GitHub Actions

mrmundt committed:
Merging work with master

carldlaird committed:
clean up documentation and add hessian example

carldlaird committed:
cleaning up tests for cyipopt external grey box examples

carldlaird committed:
removing rogue print statement

jsiirola committed:
Remove PySP console scripts and plugin registrations

mrmundt committed:
Merging with master

mrmundt committed:
Reworking awkward standard_repn usage of isclose

mrmundt committed:
Correcting GAMS

carldlaird committed:
responding to PR review

mrmundt committed:
Marking potential bug

mrmundt committed:
Merge remote-tracking branch 'origin' into pyutilib-misc

mrmundt committed:
Merge remote-tracking branch 'origin' into jenkins_fix

mrmundt committed:
Removing check for Python lt 3.6

mrmundt committed:
Merge pull request #1809 from mrmundt/pyutilib-math

Remove pyutilib.math Dependencies

mrmundt committed:
Removing unnecessary line

mrmundt committed:
Reworking several instances of 'shell=True' subprocess calls

mrmundt committed:
Recreating pyutilib's run_file in test_tutorials

mrmundt committed:
Reverting run_file work - moved to another branch

mrmundt committed:
Reworking to minimize repetitive code

mrmundt committed:
Forgot to split into a list

jsiirola committed:
Removing special handling of max problems in NEOS/RAPOSa

mrmundt committed:
Merge pull request #1816 from jsiirola/neos-rapose-fix

Track changes in NEOS/RAPOSa expected behavior

mrmundt committed:
Merge branch 'master' into pyutilib-runfile

mrmundt committed:
Merge branch 'master' into pyutilib-misc

mrmundt committed:
Merge branch 'master' into pyutilib-subprocess

blnicho committed:
Merge pull request #1810 from carldlaird/external-interface-hessians

External interface hessians

mrmundt committed:
Adding some requested changes

mrmundt committed:
Merging changes

mrmundt committed:
Replacing instances of stderr=subprocess.PIPE with STDOUT

mrmundt committed:
Adding import_file tests

mrmundt committed:
Reverting pysp changes

mrmundt committed:
Merge pull request #1814 from mrmundt/pyutilib-runfile

Remove pyutilib.misc.run_file

mrmundt committed:
Adding bug fix

mrmundt committed:
Changing implementation of virtualenv

mrmundt committed:
Merge pull request #1813 from mrmundt/jenkins_fix

Jenkins Script Correction for Python 3.6+

mrmundt committed:
Raising exception if one is hit in import_file

mrmundt committed:
Removed unnecessary print statement

mrmundt committed:
Changing exception

mrmundt committed:
Raising new error if module = None

mrmundt committed:
Printing relevant information in catch

mrmundt committed:
Adding one more piece of relevant info

mrmundt committed:
Adding .py to each problem in piecewise_linear tests

mrmundt committed:
Adding a path check and a test to assure appropriate failure

mrmundt committed:
Adding requested changes

blnicho committed:
Merge pull request #1815 from mrmundt/pyutilib-misc

Replace pyutilib.misc.import_file

carldlaird committed:
initial working of nlp projections

carldlaird committed:
removing array_equal when there are nans

carldlaird committed:
reverting numpy requirement increase

mrmundt committed:
yaml_fix, quote_split, and tostr all replaced

mrmundt committed:
Updating

mrmundt committed:
Different invocation

mrmundt committed:
Trying another approach

carldlaird committed:
Merge branch 'master' of https://github.com/pyomo/pyomo into multiple-external-grey-box

mrmundt committed:
Found the issue

jsiirola committed:
Adding TeeStream (and tests)

mrmundt committed:
Merge branch 'pyutilib-subprocess' of github.com:mrmundt/pyomo into pyutilib-subprocess

jsiirola committed:
Adding a TeeStream.__del__ method

jsiirola committed:
Add support for universal newlines and user-specified encodings

jsiirola committed:
Add delay to tests to improve determinism.

jsiirola committed:
Additional delays to improve determinism

jsiirola committed:
Improve merge test determinism

jsiirola committed:
Additional refinement of encoding, test management

jsiirola committed:
Resolve issue ignoring streams added after _mergedReader started

mrmundt committed:
Merge branch 'pyutilib-subprocess' of github.com:mrmundt/pyomo into pyutilib-subprocess

mrmundt committed:
Changing shellcmd to include new TeeStream

mrmundt committed:
Applying new TeeStream

mrmundt committed:
Logic oops

mrmundt committed:
Adding universal_newlines and removing decodes

mrmundt committed:
GAMS tee reimplementation with TeeStream

jsiirola committed:
Break lines using the current setting for _StreamHandle.newlines

jsiirola committed:
encode() not necessary with universal_newlines

jsiirola committed:
Use Popen when sending input to stdin

mrmundt committed:
Removing print statement and result =

jsiirola committed:
Adding timeout to TeeStream join

jsiirola committed:
Cleanup formatting; subprocess.run() input takes a str not bytes

jsiirola committed:
Switch back to using subprocess.run(), set universal_newlines=True

jsiirola committed:
Avoid raising a deadlock exception when processing another exception

mrmundt committed:
Applying requested changes

jsiirola committed:
Extend the timeout when joining TeeIO threads

jsiirola committed:
Additional debugging when catching an exception while creating solver

jsiirola committed:
Adding missing universal_newlines flag

jsiirola committed:
debugging: adding handle state

jsiirola committed:
Add lock to prevent closing file during select

jsiirola committed:
Pass select() a copy of the handle list

jsiirola committed:
Update handle state debugging information

jsiirola committed:
Bug fix: missing var definition

jsiirola committed:
Additional debugging

jsiirola committed:
more debugging for handle deadlock

jsiirola committed:
Change the order of handler creation

jsiirola committed:
Explicitly manage the file and file descriptor

jsiirola committed:
debugging: add indicator for handle activity

jsiirola committed:
Buffer handle list when closing threads

jsiirola committed:
Removing debugging

jsiirola committed:
Increase delay to improve determinism

jsiirola committed:
Update expected number of codecov reports

jsiirola committed:
Fix typo

jsiirola committed:
Increase delay to improve determinism (OSX)

jsiirola committed:
Merge pull request #1818 from jsiirola/codecov-update

Update expected number of codecov reports

jsiirola committed:
Merge branch 'master' into remove-pysp

mrmundt committed:
Implementing requested changes

mrmundt committed:
Merge branch 'master' into pyutilib-subprocess

jsiirola committed:
Remove PySP documentation from Pyomo

jsiirola committed:
Removing PySP and PySP examples

jsiirola committed:
Removing the phpyro solver manager

jsiirola committed:
Removing references to PySP code

jsiirola committed:
Updating references to pysp modules

jsiirola committed:
Adding deprecation wrapper for pyomo.pysp

jsiirola committed:
Updating test comments/docs

jsiirola committed:
Suppress pysp ImportError is nose is running

jsiirola committed:
Skip parmest doctests if parmest is not available

jsiirola committed:
Remove 'parallel' GHA jobs (only used by PySP)

mrmundt committed:
Merge pull request #1812 from mrmundt/pyutilib-subprocess

Replace pyutilib.subprocess with Python subprocess

jsiirola committed:
Only implement the pyomo.pysp deprecation path if nose is not running

mrmundt committed:
Merging with subprocess change

carldlaird committed:
working on _ExternalGreyBoxAsNLP tests

jsiirola committed:
Replacing PySP reference with mpi-sppy; reformatting

carldlaird committed:
completed implementation and testing of _ExternalGreyBoxAsNLP

mrmundt committed:
Merge pull request #1819 from jsiirola/remove-pysp

Remove PySP

mrmundt committed:
Adding redirect work - including upgrades to TeeStream

mrmundt committed:
Merge branch 'master' into pyutilib-misc

mrmundt committed:
Finishing redirect transfer

mrmundt committed:
capture_output must close before it can be compared

jsiirola committed:
Add missing __init__.py from tests directories

jsiirola committed:
Put qt tests in a TestCase class so test.pyomo picks them up

jsiirola committed:
Fix argument typo

mrmundt committed:
Creating flattener and reworking all uses of flatten*

mrmundt committed:
Updating comments and docstrings

mrmundt committed:
Removing comment

mrmundt committed:
Removing unnecessary newline

jsiirola committed:
Define pyomo.common.unittest.timeout test decorator

jsiirola committed:
Switch QT tests to directly call pytest.main()

jsiirola committed:
Catch implicit contextvars dependency

jsiirola committed:
Add unittests for unittest.timeout

jsiirola committed:
Adding documentation

jsiirola committed:
Update coverage to track multiprocessing

mrmundt committed:
Adding StreamIndenter and flattener tests

jsiirola committed:
Avoid pickling function, args for timeout runner

jsiirola committed:
Skip timeout tests on platforms without multiprocessing==fork

mrmundt committed:
Incorporating requested changes

mrmundt committed:
Forgot to update indent size on indent tests

mrmundt committed:
What happens if we get rid of unicode

mrmundt committed:
Adding -C SHA flag to codecov-upload

mrmundt committed:
Removing basestring instances

mrmundt committed:
Removing string_types

mrmundt committed:
Removing more Python2 specifics and fixing a problem

jsiirola committed:
Merge pull request #1822 from mrmundt/pyutilib-misc

Replace miscellaneous pyutilib.misc items

jsiirola committed:
Get the SHA from the PR HEAD, if applicable

jsiirola committed:
Resolve differences between GHA drivers

jsiirola committed:
Add implied success checks for Codecov uploads

jsiirola committed:
Merge pull request #1824 from mrmundt/codecov_bug

Manually add commit HASH to codecov upload on GitHub Actions

blnicho committed:
Merge pull request #1820 from jsiirola/viewer-tests

Update contrib.viewer tests

jsiirola committed:
Overhaul timeout to work on Windows/OSX

carldlaird committed:
support for multiple external grey box models

carldlaird committed:
Merge remote-tracking branch 'main-pyomo/master' into multiple-external-grey-box

jsiirola committed:
Merge branch 'master' into viewer-tests

jsiirola committed:
Capture and report stdout in unittest.timeout()

carldlaird committed:
fixed hessian computation

carldlaird committed:
fixing dynamic model test to use new external grey box code

jsiirola committed:
Merge pull request #1825 from jsiirola/viewer-tests

Update unittest.timeout to work on Windows

mrmundt committed:
Merging with master

michaelbynum committed:
benders rename master to root

michaelbynum committed:
benders rename master to root

blnicho committed:
Merge pull request #1828 from michaelbynum/benders

Benders: rename master problem to root problem

ghackebeil committed:
removing use of six from kernel

jsiirola committed:
Relax tee/timeout test timing to improve robustness

jsiirola committed:
Fix typo introduced in 747f0c9

blnicho committed:
Merge pull request #1831 from jsiirola/relax-test-timing

Relax tee/timeout test timing to improve robustness

blnicho committed:
Merge pull request #1829 from Pyomo/goodbye_py2_cleanup

Cleanup of code related to Python2/3 cross-compatibility (in kernel)

blnicho committed:
Merge branch 'master' into standardize_string

blnicho committed:
Removing old, unreachable code

carldlaird committed:
implementation and tests for duals

mrmundt committed:
First round of six conversion

mrmundt committed:
Somehow missed a parenthesis

mrmundt committed:
Missed a six statement

mrmundt committed:
Adding util changes

mrmundt committed:
Pyomo.core six changes

mrmundt committed:
Typo

mrmundt committed:
Next round of core work - unsure of builtins will work

mrmundt committed:
Needed to import builtins

mrmundt committed:
Typo

mrmundt committed:
Last round of pyomo core six fixes

carldlaird committed:
added parameter estimation example for external grey box models

carldlaird committed:
adding tests and some README documentation

carldlaird committed:
adding missing __init__.py

carldlaird committed:
fixing tests related to addition of exception when the NL has no variables

mrmundt committed:
Misplaced values for items

mrmundt committed:
Missed one iteritems

mrmundt committed:
Reverting 5 files not in core

mrmundt committed:
One extra file revert

mrmundt committed:
Integrating suggested changes

jsiirola committed:
Merge pull request #1826 from mrmundt/standardize_string

Standardize usage of string type comparison

mrmundt committed:
Adding requested changes and resolving merge conflicts

mrmundt committed:
Somehow missed a file

mrmundt committed:
Attempting to resolve errors

mrmundt committed:
Attempting to resolve issues, part 2

mrmundt committed:
Missed one more incorrect method call

mrmundt committed:
One final instance - I hope

carldlaird committed:
responding to comments and adding nlp return argument

mrmundt committed:
Missed one request

carldlaird committed:
adding testing for the data generator

carldlaird committed:
removing unnecessary file

Utkarsh-Detha committed:
Fixed quadratic term handling for QCQO problems in mosek_direct interface.

carldlaird committed:
Merge pull request #1832 from carldlaird/multiple-external-grey-box

Multiple external grey box models in single Pyomo formulation

jsiirola committed:
Merge pull request #1833 from mrmundt/remove-six

Replace `six` usage in pyomo.core

blnicho committed:
Merge pull request #1835 from Utkarsh-Detha/master

Fixed quadratic term handling for QCQO problems in mosek_direct interface
